modules/routes/Forums/forum_route.py

```diff
@@ -0,0 +1,91 @@
+from flask import request, jsonify
+from app import app
+from modules.models.account import AccountType
+from modules.routes.Forums.forum_schema import (
+    ForumSchema,
+    DiscussionThreadSchema,
+    DiscussionReplySchema,
+    NewDiscussionThreadSchema,
+    NewDiscussionReplySchema,
+
+)
+from modules.utils.db import db
+from modules.utils.route_utils import authenticate, fetch_session, protected_route
+
+@app.route("/forums/course/<string:course_code>", methods=["GET"])
+def get_course_forums(course_code):
+    db_cursor = db.cursor()
+    db_cursor.execute("SELECT * FROM DiscussionForum WHERE course_code = %s", (course_code,))
+    forums = db_cursor.fetchall()
+    if not forums:
+        return jsonify({"message": "No forums found for this course"}), 404
+    # Convert the forums data into JSON format and return
+    return jsonify(forums), 200
+
+@app.route("/forums/course/<string:course_code>", methods=["POST"])
+@protected_route(roles=[AccountType.Admin, AccountType.Lecturer])
+def create_course_forum(course_code):
+    body = ForumSchema().load(request.get_json(force=True))
```

Comment:
Add checks to see if the user is a student using the `fetch_session()` method. If they are a student, they should not have access to this route. If they are a lecturer, perform a check to see if they teach this course. If they do not, they should not have access to this route. If all checks pass, proceed with logic.